allennlp/data/dataset_readers/stanford_sentiment_tree_bank.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, List
+import logging
+
+from overrides import overrides
+from nltk.tree import Tree
+import tqdm
+
+from allennlp.common import Params
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import LabelField, TextField, Field
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.common.checks import ConfigurationError
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("sst_tokens")
+class StanfordSentimentTreeBankTokensDatasetReader(DatasetReader):
+    """
+    Reads tokens and their sentiment labels from the Stanford Sentiment Treebank.
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We use this to define the input representation for the text.  See :class:`TokenIndexer`.
+        Note that the `output` tags will always correspond to single token IDs based on how they
+        are pre-tokenised in the data file.
+    use_subtrees : ``bool``, optional, (default = ``False``)
+        Whether or not to use sentiment-tagged subtrees.
+    granularity : ``str``, optional (default = ``"5-class"``)
+        One of ``"5-class"``, ``"3-class"``, or ``"2-class"``, indicating the number
+        of sentiment labels to use. The Stanford Sentiment Treebank comes with labels
```

Comment:
You might want to move this discussion into the part of the docstring where you talk about file formats, above, as this seems like it shouldn't be buried in a parameter description.  Up to you, though.